synapseclient/models/activity.py

```diff
@@ -1 +1,345 @@
+        self.created_by = synapse_activity.get("createdBy", None)
+        self.modified_by = synapse_activity.get("modifiedBy", None)
+        self.executed = []
+        self.used = []
+        for used in synapse_activity.get("used", []):
+            concrete_type = used.get("concreteType", None)
+            if USED_URL_CONCRETE_TYPE == concrete_type:
+                used_url = UsedURL(
+                    name=used.get("name", None),
+                    url=used.get("url", None),
+                )
+
+                if used.get("wasExecuted", False):
+                    self.executed.append(used_url)
+                else:
+                    self.used.append(used_url)
+            elif USED_ENTITY_CONCRETE_TYPE == concrete_type:
+                reference = used.get("reference", {})
+                used_entity = UsedEntity(
+                    target_id=reference.get("targetId", None),
+                    target_version_number=reference.get("targetVersionNumber", None),
+                )
+
+                if used.get("wasExecuted", False):
+                    self.executed.append(used_entity)
+                else:
+                    self.used.append(used_entity)
+
+        return self
+
+    def _create_used_and_executed_synapse_activities(
+        self,
+    ) -> Tuple[List[Dict], List[Dict]]:
+        """
+        Helper function to create the used and executed activities for the Synapse Activity.
+
+        Returns:
+            A tuple of the used and executed activities.
+        """
+        synapse_activity_used = []
+        synapse_activity_executed = []
+
+        for used in self.used:
+            if isinstance(used, UsedEntity):
+                synapse_activity_used.append(
+                    {
+                        "reference": {
+                            "targetId": used.target_id,
+                            "targetVersionNumber": used.target_version_number,
+                        }
+                    }
+                )
+            elif isinstance(used, UsedURL):
+                synapse_activity_used.append(
+                    {
+                        "name": used.name,
+                        "url": used.url,
+                    }
+                )
+
+        for executed in self.executed:
+            if isinstance(executed, UsedEntity):
+                synapse_activity_executed.append(
+                    {
+                        "reference": {
+                            "targetId": executed.target_id,
+                            "targetVersionNumber": executed.target_version_number,
+                        },
+                        "wasExecuted": True,
+                    }
+                )
+            elif isinstance(executed, UsedURL):
+                synapse_activity_executed.append(
+                    {"name": executed.name, "url": executed.url, "wasExecuted": True}
+                )
+        return synapse_activity_used, synapse_activity_executed
```

Comment:
Nit: not to make things more complex, but tuples are irritating at times when keeping order.  What are your thoughts on `namedtuples`?  (Note: I'm good with this, just opening a discussion.)

Reply:
I like the idea of namedtuples - I added this in https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1055/commits/a38f2c2bfa771afdd0dad03a9382bce311b98ce4